stable/grafana/templates/configmap.yaml

```diff
@@ -1,15 +1,52 @@
 apiVersion: v1
 kind: ConfigMap
 metadata:
+  name: {{ template "grafana.fullname" . }}
   labels:
-    app: {{ template "grafana.fullname" . }}
-    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
-    component: "{{ .Values.server.name }}"
-    heritage: "{{ .Release.Service }}"
-    release: "{{ .Release.Name }}"
-  name: {{ template "grafana.server.fullname" . }}-config
+    app: {{ template "grafana.name" . }}
+    chart: {{ template "grafana.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
 data:
-  {{- if .Values.server.installPlugins }}
-  grafana-install-plugins: {{ .Values.server.installPlugins | quote }}
+{{- if .Values.plugins }}
+  plugins: {{ default "" .Values.plugins | quote }}
+{{- end }}
+  grafana.ini: |
+{{- range $key, $value := index .Values "grafana.ini" }}
+    [{{ $key }}]
+    {{- range $elem, $elemVal := $value }}
+    {{ $elem }} = {{ $elemVal }}
+    {{- end }}
+{{- end }}
+
+{{- if .Values.datasources }}
+  {{- range $key, $value := .Values.datasources }}
+  {{ $key }}: |
+{{ toYaml $value | indent 4 }}
+  {{- end -}}
+{{- end -}}
```

Comment:
The datasources configuration can contain passwords and other sensitive information. Seems like it should go into a Secret instead of a ConfigMap.

Reply:
Good call.  grafana.ini should probably also be a secret.  Will fix in follow-up PR.